tests/bdd/bdd_test.go

```diff
@@ -0,0 +1,247 @@
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	scheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var kubeconfig = string(os.Getenv("HOME") + "/.kube/config")
+var config, _ = clientcmd.BuildConfigFromFlags("", kubeconfig)
+var client, _ = kubernetes.NewForConfig(config)
+var clientSet, _ = clientV1alpha1.NewForConfig(config)
+
+func TestChaos(t *testing.T) {
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "BDD test")
+}
+
+var _ = BeforeSuite(func() {
+
+	err := v1alpha1.AddToScheme(scheme.Scheme)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	//Creating crds
+	By("creating chaosengine crd")
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/crds/chaosengine_crd.yaml").Run()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	By("creating chaosexperiment crd")
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/crds/chaosexperiment_crd.yaml").Run()
+
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	By("creating chaosresult crd")
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/crds/chaosresults_crd.yaml").Run()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	//creating rbacs
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/rbac.yaml").Run()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	//creating opearator
+
+	By("creating operator")
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/operator.yaml").Run()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	fmt.Println("operator created successfully")
+
+})
+
+//BDD Tests
+var _ = Describe("BDD on chaos-operator", func() {
+
+	// BDD TEST CASE 1
+
+	Context("Check for the custom resources", func() {
+
+		It("chaosengine Runner pod should present", func() {
+
+			//creating nginx deployment
+
+			deployment := &appv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "nginx",
+					Labels: map[string]string{
+						"app": "nginx",
+					},
+					Annotations: map[string]string{
+						"litmuschaos.io/chaos": "true",
+					},
+				},
+				Spec: appv1.DeploymentSpec{
+					Replicas: func(i int32) *int32 { return &i }(3),
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"app": "nginx",
+						},
+					},
+					Template: v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"app": "nginx",
+							},
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Name:  "nginx",
+									Image: "nginx:latest",
+									Ports: []v1.ContainerPort{
+										{
+
+											ContainerPort: 80,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			_, err := client.AppsV1().Deployments("default").Create(deployment)
+			if err != nil {
+				panic(err)
```

Comment:
Can you generate some error here with error message

Reply:
okay